app/components/table.gts

```diff
@@ -5,114 +5,37 @@ import {
-          viewBox="0 0 24 24"
-          data-icon="SvgShoppingCart"
-          aria-hidden="true"
-          className="w-4 h-4"
-        >
-          <path
-            d="M20.112 19.4a1.629 1.629 0 11-1.629-1.629 1.63 1.63 0 011.629 1.629zM9.941 17.768a1.629 1.629 0 101.628 1.632 1.629 1.629 0 00-1.628-1.632zM3 3.006h1.678a2.113 2.113 0 011.965 1.573l2.051 9.152a2.114 2.114 0 001.965 1.574h6.788a2.153 2.153 0 001.989-1.568L20.957 8.8a1.233 1.233 0 00-1.236-1.588L11.4 7.064"
-            fill="none"
-            strokeLinecap="round"
-            strokeLinejoin="round"
-            strokeWidth="2"
-          ></path>
-        </svg>
-        <span>Buy now</span>
-    `;
-
-    this.eventListener = () => alert(
-          `You selected "${
-            params.data?.title
-          }" to purchase for ${Intl.NumberFormat("us", {
-            style: "currency",
-            currency: "USD",
-          }).format(params.data?.price)}`
-        );
-    this.eGui.addEventListener('click', this.eventListener);
-  }
-
-  // Mandatory - Return the DOM element of the component, this is what the grid puts into the cell
-  getGui() {
-    return this.eGui;
-  }
-
-  // Optional - Gets called once by grid after rendering is finished - if your renderer needs to do any cleanup,
-  // do it here
-  destroy() {
-    if (this.eGui) {
-      this.eGui.removeEventListener('click', this.eventListener);
-    }
-  }
-
-  // Mandatory - Get the cell to refresh. Return true if the refresh succeeded, otherwise return false.
-  // If you return false, the grid will remove the component from the DOM and create
-  // a new component in its place with the new values.
-  refresh(params: ICellRendererParams) {
-    // set value into cell again
-    this.cellValue = this.getValueToDisplay(params);
-    this.eValue.innerHTML = this.cellValue;
-
-    // return true to tell the grid we refreshed successfully
-    return true;
-  }
+import {
+  GridApi
+} from 'ag-grid-enterprise';
 
-  getValueToDisplay(params: ICellRendererParams) {
-    return params.valueFormatted ? params.valueFormatted : params.value;
-  }
+import EmberCellRenderer from 'polaris-starter/utils/ember-cell-renderer';
+import BuyNow from './buy-now'
+class BuyButtonCellRenderer extends EmberCellRenderer {
+  component = BuyNow;
 }
 
-/**
- * Renders a custom tooltip for the Description column. I wanted to see how easy it was
- * to render custom tooltip components.
- */
-class DescriptionTooltip implements ITooltipComp {
-  eGui!: HTMLDivElement;
-
-  init(params: ITooltipParams) {
-    this.eGui = document.createElement('div');
-    this.eGui.className = 'bg-slate-600 text-white rounded-md p-2';
-    this.eGui.innerHTML = `${params.data?.description}`;
-  }
-
-  getGui() {
-    return this.eGui;
-  }
+import EmberTooltipRenderer from 'polaris-starter/utils/ember-tooltip-renderer';
+import Tooltip from './tooltip'
+class TooltipRenderer extends EmberTooltipRenderer {
+  component = Tooltip;
```

Comment:
Glint/TS looks to be unhappy with this `component` property - should we type it properly or do we not care for this demo?